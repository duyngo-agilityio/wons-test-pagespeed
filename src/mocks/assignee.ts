import { TRole } from '@/types';

export const MOCK_ASSIGNEES = {
  data: [
    {
      id: 1,
      attributes: {
        name: 'John Doe',
        avatar: '/avatar1.jpg',
        email: 'john@example.com',
        username: 'john_doe',
        fullName: 'John Doe',
        password: 'password123',
        role: 'User' as unknown as TRole,
        token: 'token123',
      },
    },
    {
      id: 2,
      attributes: {
        name: 'Jane Smith',
        avatar: '/avatar2.jpg',
        email: 'jane@example.com',
        username: 'jane_smith',
        fullName: 'Jane Smith',
        password: 'password123',
        role: 'User' as unknown as TRole,
        token: 'token123',
      },
    },
  ],
};
