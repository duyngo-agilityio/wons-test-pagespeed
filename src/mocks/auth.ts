// Types
import { ISignUpFormData } from '@/types';

export const SIGN_UP_FORM_DATA_MOCK: ISignUpFormData = {
  fullName: 'John Doe',
  username: 'johndoe',
  email: 'john.doe@example.com',
  password: 'Abcd@0987',
};

export const MOCK_SUCCESS_SIGN_UP_RES = {
  data: { user: { ...SIGN_UP_FORM_DATA_MOCK, id: '1' } },
  totalCount: 0,
};
