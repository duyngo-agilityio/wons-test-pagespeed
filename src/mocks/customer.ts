// Models
import { ICustomer } from '@/models';

// Types
import { TableColumnType } from '@/types';

export const PRODUCT_TABLE_COLUMNS_MOCK: TableColumnType<Partial<ICustomer>>[] =
  [
    {
      header: 'Name',
      accessor: 'name',
      isSort: true,
    },
    {
      header: 'FirstName',
      accessor: 'firstName',
      isSort: true,
    },
    {
      header: 'LastName',
      accessor: 'lastName',
      isSort: true,
    },
    {
      header: 'Email',
      accessor: 'email',
      isSort: true,
    },
  ];

export const PRODUCT_MOCK = [
  {
    id: '1',
    name: 'John Doe',
    firstName: 'John',
    lastName: 'Doe',
    email: 'john.doe@example.com',
    phone: '123-456-7890',
    job: 'Software Engineer',
    address: '123 Main St, City, Country',
    avatar: 'https://example.com/avatar1.jpg',
  },
  {
    id: '2',
    name: 'Jane Smith',
    firstName: 'Jane',
    lastName: 'Smith',
    email: 'jane.smith@example.com',
    phone: '098-765-4321',
    job: 'Project Manager',
    address: '456 Elm St, City, Country',
    avatar: 'https://example.com/avatar2.jpg',
  },
  {
    id: '3',
    name: 'Alice Johnson',
    firstName: 'Alice',
    lastName: 'Johnson',
    email: 'alice.johnson@example.com',
    phone: '555-666-7777',
    job: 'UX Designer',
    address: '789 Oak St, City, Country',
    avatar: 'https://example.com/avatar3.jpg',
  },
  {
    id: '4',
    name: 'Bob Williams',
    firstName: 'Bob',
    lastName: 'Williams',
    email: 'bob.williams@example.com',
    phone: '111-222-3333',
    job: 'Data Scientist',
    address: '101 Pine St, City, Country',
    avatar: 'https://example.com/avatar4.jpg',
  },
  {
    id: '5',
    name: 'Charlie Brown',
    firstName: 'Charlie',
    lastName: 'Brown',
    email: 'charlie.brown@example.com',
    phone: '444-555-6666',
    job: 'Marketing Specialist',
    address: '202 Birch St, City, Country',
    avatar: 'https://example.com/avatar5.jpg',
  },
];
