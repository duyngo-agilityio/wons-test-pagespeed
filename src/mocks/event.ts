export const EVENT_MOCKS = {
  title: 'Test Event',
  status: 'active',
  location: 'New York',
  date: new Date('2024-10-14'),
  startTime: '10:00',
  endTime: '12:00',
  users_permissions_users: [1, 2],
};

export const EVENTS_MOCKS = [
  {
    id: 1,
    attributes: {
      title: 'Test Event',
      status: 'active',
      location: 'New York',
      date: new Date('2024-10-14'),
      startTime: '10:00',
      endTime: '12:00',
      users_permissions_users: [1, 2],
    },
  },

  {
    id: 2,
    attributes: {
      title: 'Implement Component',
      descriptions: 'Implement for all component',
      date: '2024-10-20T01:30:00Z',
      time: '2024-10-20T01:30:00Z',
      createdAt: '2024-10-22T09:46:11.546Z',
      updatedAt: '2024-10-22T09:46:16.769Z',
      publishedAt: '2024-10-22T09:46:16.768Z',
    },
  },
];

export const EVENTS_MOCKS_WIDTH_USERS_PERMISSIONS = {
  title: 'Test Event',
  status: 'active',
  location: 'New York',
  date: new Date('2024-10-14'),
  startTime: '10:00',
  endTime: '12:00',
  users_permissions_users: {
    data: [
      {
        id: 3,
        attributes: {
          username: 'superadmin',
          email: 'admin1@gmail.com',
          provider: 'local',
          confirmed: false,
          blocked: false,
          fullName: 'Super Admin',
          createdAt: '2024-08-28T04:29:52.000Z',
          updatedAt: '2024-10-11T02:14:38.208Z',
          avatar:
            'https://watermark.lovepik.com/photo/20211209/large/lovepik-japanese-fresh-girl-park-photo-picture_501698500.jpg',
        },
      },
    ],
  },
};
