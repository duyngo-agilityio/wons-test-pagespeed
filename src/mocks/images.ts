export const MOCK_IMAGES = [
  'https://media.wired.com/photos/6500ad57fe61eb702d721b58/1:1/w_1800,h_1800,c_limit/Apple-iPhone-15-Pro-Hero-Gear.jpg',
  'https://clickbuy.com.vn/uploads/images/2019/09/thumb_iP11_3.jpg',
];
