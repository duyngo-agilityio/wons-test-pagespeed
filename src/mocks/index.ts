export * from './customer';
export * from './recentServices';
export * from './tabs';
