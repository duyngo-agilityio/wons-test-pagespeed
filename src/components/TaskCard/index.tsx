'use client';

import Image from 'next/image';

// Mocks
import { MOCK_USERS } from '@/mocks';

// Components
import LevelCardComponent from '../LevelCard';
import DropdownActions from '../DropdownActions';
import AvatarGroup from '../AvatarGroup';

// Constants
import { BLUR_SRC } from '@/constants';

type TTaskCardProps = {
  title: string;
  images?: string[];
  description?: string;
};

const TaskCard = ({ title, images, description }: TTaskCardProps) => {
  // TODO:: Handle later
  const handleDelete = () => {};

  // TODO:: Handle later
  const handleEdit = () => {};

  return (
    <div className="w-[275px] bg-white dark:bg-gray-400 p-[20px] rounded-5">
      <div className="flex flex-row items-center justify-between mb-[15px]">
        <div className="text-md">{title}</div>
        <DropdownActions
          id={1}
          onDelete={handleDelete}
          onEdit={handleEdit}
          isIconOnly
          disableAnimation
          customClassName="w-[15px] min-w-[15px]"
        />
      </div>
      <LevelCardComponent />
      <div className="mt-[20px] text-sm">{description}</div>
      {/* If images exists and has exactly one item */}
      {images && images.length === 1 && (
        <div className="mt-[20px]">
          <Image
            alt={MOCK_USERS[0].fullName}
            src={images[0]}
            width={235}
            height={176}
            placeholder="blur"
            blurDataURL={BLUR_SRC.DEFAULT}
          />
        </div>
      )}
      {/* If images exists and has exactly two items */}
      {images && images.length === 2 && (
        <div className="flex flex-row justify-between w-[235px] mt-[20px]">
          {images.map((item, index) => (
            <Image
              key={index}
              alt="Image Task"
              src={item}
              width={107}
              height={90}
              style={{ borderRadius: '10px' }}
              placeholder="blur"
              blurDataURL={BLUR_SRC.DEFAULT}
            />
          ))}
        </div>
      )}
      <div className="mt-[20px]">
        <AvatarGroup users={MOCK_USERS} />
      </div>
    </div>
  );
};

export default TaskCard;
