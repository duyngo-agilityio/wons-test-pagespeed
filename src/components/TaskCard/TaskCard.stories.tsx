// Libs
import type { Meta, StoryObj } from '@storybook/react';

// components
import { TaskCard } from '@/components';

// Mocks
import { MOCK_USERS } from '@/mocks';

const meta: Meta<typeof TaskCard> = {
  title: 'Components/TaskCard',
  component: TaskCard,
  argTypes: {
    title: {
      description:
        'The title of the task, displayed prominently at the top of the card.',
      control: { type: 'text' },
    },
    description: {
      description:
        'A brief description of the task, providing context or details about the task.',
      control: { type: 'text' },
    },
    images: {
      description:
        'An array of image URLs to display avatars or related icons. It can render one or two images depending on the number of items in the array.',
      control: { type: 'text' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof TaskCard>;

export const Default: Story = {
  args: {
    title: 'Dashboard Design',
    description: 'Discussion for management dashboard ui design',
    images: [],
  },
};

export const OneImage: Story = {
  args: {
    title: 'Landing page Design',
    description: 'Discussion for management dashboard ui design',
    images: [MOCK_USERS[0].avatar],
  },
};
export const TwoImages: Story = {
  args: {
    title: 'Dashboard Design',
    description: 'Discussion for management dashboard ui design',
    images: [MOCK_USERS[0].avatar, MOCK_USERS[1].avatar],
  },
};
