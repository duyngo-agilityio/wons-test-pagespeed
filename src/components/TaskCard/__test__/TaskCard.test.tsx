import { render, screen } from '@testing-library/react';

// Mocks
import { MOCK_USERS } from '@/mocks';

// Components
import TaskCard from '..';

// eslint-disable-next-line react/display-name
jest.mock('next/image', () => ({ src, alt }: { src: string; alt: string }) => (
  <img src={src} alt={alt} />
));

// Snapshot and other tests
describe('TaskCard Component', () => {
  const defaultProps = {
    title: 'Demo Task',
    description: 'This is a demo task description',
  };

  it('matches the snapshot', () => {
    const { container } = render(<TaskCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('renders with one image correctly', () => {
    const images = [MOCK_USERS[0].avatar];
    render(<TaskCard {...defaultProps} images={images} />);

    const image = screen.getByAltText(MOCK_USERS[0].fullName);
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', images[0]);
  });

  it('renders title and description correctly', () => {
    render(<TaskCard {...defaultProps} />);

    expect(screen.getByText('Demo Task')).toBeInTheDocument();
    expect(
      screen.getByText('This is a demo task description'),
    ).toBeInTheDocument();
  });

  it('renders with two images correctly', () => {
    const images = [MOCK_USERS[0].avatar, MOCK_USERS[1].avatar];
    render(<TaskCard {...defaultProps} images={images} />);

    const imageElements = screen.getAllByAltText('Image Task');
    expect(imageElements.length).toBe(2);
    expect(imageElements[0]).toHaveAttribute('src', images[0]);
    expect(imageElements[1]).toHaveAttribute('src', images[1]);
  });
});
