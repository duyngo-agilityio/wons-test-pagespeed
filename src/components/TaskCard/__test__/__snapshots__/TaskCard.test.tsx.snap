// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard Component matches the snapshot 1`] = `
<div>
  <div
    class="w-[275px] bg-white dark:bg-gray-400 p-[20px] rounded-5"
  >
    <div
      class="flex flex-row items-center justify-between mb-[15px]"
    >
      <div
        class="text-md"
      >
        Demo Task
      </div>
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-2 rounded-medium px-0 !gap-0 !transition-none data-[pressed=true]:scale-100 text-default-foreground h-10 data-[hover=true]:opacity-hover z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased border-none bg-bone w-[15px] min-w-[15px]"
        data-slot="trigger"
        data-testid="actions-btn"
        id="react-aria-:r0:"
        type="button"
      >
        <svg
          class="text-blue-800/30 dark:text-white/30"
          fill="currentColor"
          height="14"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
          />
        </svg>
      </button>
    </div>
    <div
      class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap text-small text-default-foreground px-[22px] h-[45px] max-h-[45px] rounded-[22px] bg-blue-500"
    >
      <span
        class="flex-1 font-normal px-2 text-center text-md capitalize text-white dark:text-gray-600"
      >
        medium
      </span>
    </div>
    <div
      class="mt-[20px] text-sm"
    >
      This is a demo task description
    </div>
    <div
      class="mt-[20px]"
    >
      <div
        class="flex items-center"
      >
        <div
          class="relative w-7 h-7 rounded-full "
        >
          <img
            alt="user1"
            src="https://img-cdn.pixlr.com/image-generator/history/65bb506dcb310754719cf81f/ede935de-1138-4f66-8ed7-44bd16efc709/medium.webp"
          />
        </div>
        <div
          class="relative w-7 h-7 rounded-full -ml-3"
        >
          <img
            alt="user2"
            src="https://img-cdn.pixlr.com/image-generator/history/65bb506dcb310754719cf81f/ede935de-1138-4f66-8ed7-44bd16efc709/medium.webp"
          />
        </div>
        <div
          class="relative w-7 h-7 rounded-full -ml-3"
        >
          <img
            alt="user4"
            src="https://img-cdn.pixlr.com/image-generator/history/65bb506dcb310754719cf81f/ede935de-1138-4f66-8ed7-44bd16efc709/medium.webp"
          />
        </div>
        <button
          class="z-0 group box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default data-[hover=true]:opacity-hover border-transparent dark:text-white px-5 pb-2.5 pt-[11px] text-md rounded-[10px] min-w-max !px-0 relative flex w-7 h-7 items-center justify-center !rounded-full !bg-teal-500 dark:!bg-teal-300 text-white -ml-3 border-2 hover:!bg-teal-600 dark:hover:!bg-teal-400 transition duration-300 ease-in-out"
          type="button"
        >
          <div
            class="flex items-center"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="12"
                x2="12"
                y1="5"
                y2="19"
              />
              <line
                x1="5"
                x2="19"
                y1="12"
                y2="12"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
